### PR TITLE
feat: add `esModule` option

### LINKED ISSUE #187: Ability to emit CJS only

### PR DESCRIPTION
**Pull request by rojvv**
Closes #187

### CODE REVIEW

File: lib/package_json.test.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ Deno.test("single entrypoint", () => {
       version: "0.1.0",
     },
     testEnabled: true,
+    includeEsModule: true,
```
Comment (dsherret) [inline lib/package_json.test.ts:45]: We should add a test here that sets this to `false`.

File: mod.ts [hunk 2]
```diff
@@ -132,10 +136,14 @@ export interface BuildOptions {
 
 /** Builds the specified Deno module to an npm package using the TypeScript compiler. */
 export async function build(options: BuildOptions): Promise<void> {
+  if (options.scriptModule == false && options.esModule == false) {
```
Comment (dsherret) [inline mod.ts:139]:
```suggestion
  if (options.scriptModule === false && options.esModule === false) {
```

Comment (dsherret) [inline mod.ts:139]: Would you be able to add one test for this that just verifies this throws? That way we can ensure it doesn't break in the future.

**Review by dsherret — CHANGES_REQUESTED** [review]
Thanks! Looks good and almost there, but we should add an integration test in tests/integration.test.ts

**Comment by dsherret** [comment]
Also, I think this is missing the necessary updates to the test runner. If you add an integration test in tests/integration.test.ts you should see it fail.

File: mod.ts (hunk 2 above)
Comment (roj1512) [reply mod.ts]: Alright!

**Comment by roj1512** [comment]
@dsherret Do you mean the `runTest` function? I added the test and it passes.

**Comment by roj1512** [comment]
@dsherret lmk how's everything now

File: lib/package_json.test.ts (hunk 1 above)
Comment (roj1512) [reply lib/package_json.test.ts:45]: Done.

File: lib/npm_ignore.test.ts [hunk 3]
```diff
@@ -9,45 +9,61 @@ Deno.test("should include src directory when the source files are not necessary"
...
     expectHasSrcFolder: true,
+    includeScriptModule: true,
+    includeEsModule: true,
   });
   runTest({
     sourceMaps: "inline",
     inlineSources: true,
     expectHasSrcFolder: true,
+    includeScriptModule: true,
+    includeEsModule: true,
```
Comment (dsherret) [inline lib/npm_ignore.test.ts:57]: It seems these are always true? Maybe let's add a test for each false and then both true?

**Review by dsherret — APPROVED** [review]
Looks great! Thanks @roj1512, I will merge this later today.

File: lib/npm_ignore.test.ts (hunk 3 above)
Comment (roj1512) [reply lib/npm_ignore.test.ts:57]: Do you mean a separate copy of the whole `"should include src directory when the source files are not necessary"` or just modify the values of the current `runTest`s?

**Pull request merged**